static/js/giphy.js

```diff
@@ -0,0 +1,65 @@
+"use strict";
+
+const renderGrid = require("@giphy/js-components").renderGrid;
+const GiphyFetch = require("@giphy/js-fetch-api").GiphyFetch;
+const throttle = require("throttle-debounce").throttle;
+
+const gf = new GiphyFetch(page_params.giphy_api_key);
+let search_term = "trending";
+
+exports.renderGrid = function (targetEl) {
+    const render = () =>
+        renderGrid(
+            {
+                width: 300,
+                fetchGifs: (offset) =>
+                    gf.search(search_term, {
+                        offset,
+                        limit: 25,
+                        random_id: page_params.user_id,
+                    }),
+                columns: 3,
+                gutter: 6,
+                noLink: true,
+                hideAttribution: true,
+                onGifClick: (props) => {
+                    $("#compose_box_giphy_grid").popover("hide");
+                    $(
+                        "#compose-textarea",
+                    )[0].value += `[${search_term}](${props.images.downsized_medium.url})`;
+                },
+            },
+            targetEl,
+        );
+    const resizeRender = throttle(500, render);
+    window.addEventListener("resize", resizeRender, false);
+    const remove = render();
+    return {
+        remove: () => {
+            remove();
+            window.removeEventListener("resize", resizeRender, false);
+        },
+    };
+};
+
+let template =
+    '<div class="popover" id="giphy_grid_in_popover"><div class="arrow"></div><div class="popover-inner"><div class="search-box"><input type="text" id="giphy-search-query" class="search-query" placeholder="Search GIFs"></div><div class="popover-content"></div><div class="popover-footer"><img src="/static/images/GIPHY_attribution.png" alt="GIPHY attribution"></div></div></div>';
```

Comment:
Let's move this to handlebars template -- we try to avoid doing any material HTML in JS files as it makes it harder to audit for XSS issues etc.